From 065d8d8fd0c5da52b1659d8a697af6d2d4af3871 Mon Sep 17 00:00:00 2001
From: Jason Butler <jason_michael_butler@yahoo.com>
Date: Sat, 20 Nov 2010 14:26:30 -0500
Subject: [PATCH 04/23] Fixed ssl certificate problem with sites like
 https://webmail.us.army.mil

By default QSSLConfiguration was only accepting SSLv3 while we want to
accept all of them.

Issue: 915
---
 src/network/networkaccessmanager.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git src/network/networkaccessmanager.cpp src/network/networkaccessmanager.cpp
index 7e99d46..56fb343 100644
--- src/network/networkaccessmanager.cpp
+++ src/network/networkaccessmanager.cpp
@@ -168,6 +168,7 @@ void NetworkAccessManager::loadSettings()
     QList<QSslCertificate> ca_new = QSslCertificate::fromData(settings.value(QLatin1String("CaCertificates")).toByteArray());
     ca_list += ca_new;
     sslCfg.setCaCertificates(ca_list);
+    sslCfg.setProtocol(QSsl::AnyProtocol);
     QSslConfiguration::setDefaultConfiguration(sslCfg);
 #endif
 
@@ -327,6 +328,7 @@ void NetworkAccessManager::sslErrors(QNetworkReply *reply, const QList<QSslError
                 QList<QSslCertificate> ca_list = sslCfg.caCertificates();
                 ca_list += ca_new;
                 sslCfg.setCaCertificates(ca_list);
+                sslCfg.setProtocol(QSsl::AnyProtocol);
                 QSslConfiguration::setDefaultConfiguration(sslCfg);
                 reply->setSslConfiguration(sslCfg);
 
-- 
2.44.0

